### PR TITLE
chore: remove redundant ldflags

### PR DESCRIPTION
**Pull request by hectorj2f**
The go package comes with a default `ldflag: -w` so we can safely remove it from the list of set ldflags in the package definitions.

### CODE REVIEW

File: conftest.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ pipeline:
 
   - uses: go/build
     with:
-      ldflags: -w -s -X github.com/open-policy-agent/conftest/internal/commands.version=${{package.version}}
+      ldflags: -s -X github.com/open-policy-agent/conftest/internal/commands.version=${{package.version}}
```
Comment (xnox) [inline conftest.yaml:21]: hmmmm i think we agree to not strip symbols, that's why we used to have '-s' before in default flags and then dropped. please drop '-s' too as we want symbols tables for govulnchecker to work.

File: crossplane.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ pipeline:
 
   - uses: go/build
     with:
-      ldflags: -w
+      ldflags:
       output: crossplane
```
Comment (xnox) [inline crossplane.yaml:34]: empty ldflags key should be removed.

File: crossplane.yaml [hunk 3]
```diff
@@ -44,7 +44,7 @@ subpackages:
           deps: github.com/go-git/go-git/v5@v5.11.0
       - uses: go/build
         with:
-          ldflags: -w
+          ldflags:
           output: crank
```
Comment (xnox) [inline crossplane.yaml:48]: empty ldflags key should be removed.

File: cue.yaml [hunk 4]
```diff
@@ -34,7 +34,7 @@ pipeline:
       modroot: .
       packages: ./cmd/cue
       output: cue
-      ldflags: -w -buildid= -X cuelang.org/go/cmd/cue/cmd.version=${{package.version}}
+      ldflags: -buildid= -X cuelang.org/go/cmd/cue/cmd.version=${{package.version}}
```
Comment (xnox) [inline cue.yaml:37]: i thought passing in empty buildid= destroys information no? i would rather not pass one in, or have something meaningful like sha1 commit. but i can't remember the expected buildid format.

File: gitlab-logger.yaml [hunk 5]
```diff
@@ -21,7 +21,7 @@ pipeline:
     with:
       packages: ./cmd/gitlab-logger
       output: gitlab-logger
-      ldflags: -w
+      ldflags:
```
Comment (xnox) [inline gitlab-logger.yaml:24]: empty ldflags key should be dropped

File: go-bindata.yaml [hunk 6]
```diff
@@ -24,7 +24,7 @@ pipeline:
     with:
       packages: ./go-bindata
       output: go-bindata
-      ldflags: -w
+      ldflags:
```
Comment (xnox) [inline go-bindata.yaml:27]: drop.

File: crossplane.yaml (hunk 2 above)
Comment (hectorj2f) [reply crossplane.yaml:34]: oh! This wasn't intended.

File: trillian.yaml [hunk 7]
```diff
@@ -35,7 +35,7 @@ subpackages:
           subpackage: "true"
           packages: ./cmd/trillian_log_server
           output: trillian_log_server
-          ldflags: -w
+          ldflags:
```
Comment (xnox) [inline trillian.yaml:38]: drop

File: trillian.yaml [hunk 8]
```diff
@@ -46,7 +46,7 @@ subpackages:
           subpackage: "true"
           packages: ./cmd/trillian_log_signer
           output: trillian_log_signer
-          ldflags: -w
+          ldflags:
```
Comment (xnox) [inline trillian.yaml:49]: drop

**Review by xnox — CHANGES_REQUESTED** [review]
please tweak commented things.

**Review by xnox — APPROVED** [review]

**Pull request merged**